draft-ietf-lake-edhoc.md

```diff
@@ -839,6 +839,7 @@ message_1 = (
   SUITES_I : suites,
   G_X : bstr,
   C_I : bstr / -24..23,
+  ? PAD_1 : 1* true,
```

Comment:
Please check CDDL

Reply:
@cabo How about this?

? PAD_1 : padding,

...

padding = 1* true

Reply:
1* true is not a type, but a group.  So each true would turn up as a separate item in the sequence.
If that is OK, `* PAD_1: true` makes this more transparent.
If not, `? PAD_1: [* true]` is better.
(Note that I'm using * and not + (1*), because that allows more fine-grained control over the size of the padding.)

Reply:
And don't forget to update the copy...

Reply:
? PAD_1: [* true]
(this changes the wire format, and should probable by discussed with the implementors)

* PAD_1: true
(makes PAD_1 an item instead of a group, PAD_2, PAD_3, and PAD_4 are groups)

A third alternative would be:

message_1 = (
  METHOD : int,
  SUITES_I : suites,
  G_X : bstr,
  C_I : bstr / -24..23,
  ? PAD_1,
  ? EAD_1 : ead,
)

PAD_1 = 1* true

Reply:
Of course, they should all be the same constructions (and, no, I don't want to change the wire format).
An optional sequence of at least one sounds a bit baroque.
Oh, and why not padding at the end?

Reply:
What is Sturm und Drang with an optional sequence of at least one:
If there is none, why should there be a sequence?
Marco preferred this location of the padding, I think because it is the end of the simple EDHOC-internal processing; EAD_1 may contain multiple EAD items intended for the application.

Reply:
What you find on the wire will be multiple `true` items at the sequence level.  Why not represent this with some fidelity?